urllib3/util/retry.py

```diff
@@ -330,7 +338,7 @@ def increment(self, method=None, url=None, response=None, error=None,
 
         elif error and self._is_read_error(error):
             # Read retry?
-            if read is False:
+            if read is False or (method is not None and not self._is_method_retryable(method)):
```

Comment:
@Lukasa Given I'm at it, should I just add the method value to all uses of `retry.increment` in the tests (I currently have only added them to the tests that use a read error)?